Product/Production/Adapters/DocumentQuery_a0/src/main/java/gov/hhs/fha/nhinc/docquery/deferred/adapter/AdapterDeferredResultsOptionResponseSecured.java

```diff
@@ -26,20 +26,44 @@
 */
 package gov.hhs.fha.nhinc.docquery.deferred.adapter;
 
+import gov.hhs.fha.nhinc.docquery.deferred.impl.AdapterDeferredResultsOption;
 import gov.hhs.fha.nhinc.dq.adapterdeferredresultoptionsecured.AdapterDocQueryDeferredResultsOptionSecuredPortType;
+import gov.hhs.fha.nhinc.messaging.server.BaseService;
+import javax.annotation.Resource;
+import javax.xml.ws.WebServiceContext;
 import oasis.names.tc.ebxml_regrep.xsd.query._3.AdhocQueryResponse;
 import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryResponseType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 /**
  * Adapter webservice to process deferred documents into a deferred response option message to send back to the Initiating
  * Gateway at the endpoint specified in the initial request.
  */
+@Service
+public class AdapterDeferredResultsOptionResponseSecured extends BaseService implements AdapterDocQueryDeferredResultsOptionSecuredPortType{
 
-public class AdapterDeferredResultsOptionResponseSecured implements AdapterDocQueryDeferredResultsOptionSecuredPortType{
+    private static final Logger LOG = LoggerFactory.getLogger(AdapterDeferredResultsOptionResponseSecured.class);
+
+    @Autowired
+    AdapterDeferredResultsOption adapterImpl;
+
+    @Resource
+    private WebServiceContext context;
+
+    public void setContext(WebServiceContext context) {
```

Comment:
Question: Why do we setContext when we already define autowire here? Is that necessary?

Reply:
Redundancy for Setter injection - Agreed to remove